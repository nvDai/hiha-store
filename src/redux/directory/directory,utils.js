export const convertSnapshotToSectionsArray = snapshot => {
	const sections = snapshot.docs.map(doc => doc.data())
	return sections.sort(compare)
}

function compare(a, b) {
	return a.id - b.id
}
