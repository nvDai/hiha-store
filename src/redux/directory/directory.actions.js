import directoryActionTypes from './directory.types'

export const fetchDirectorySectionStart = () => ({
	type: directoryActionTypes.FETCH_DIRECTORY_SECTIONS_START,
})

export const fetchDirectorySectionSuccess = sections => ({
	type: directoryActionTypes.FETCH_DIRECTORY_SECTIONS_SUCCESS,
	payload: sections,
})

export const fetchDirectorySectionFailure = errorMessage => ({
	type: directoryActionTypes.FETCH_DIRECTORY_SECTIONS_FAILURE,
	payload: errorMessage,
})
