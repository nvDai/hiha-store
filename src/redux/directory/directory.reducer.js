import directoryActionTypes from './directory.types'

const INITIAL_STATE = {
	sections: null,
	errorMessage: undefined,
}

const directoryReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {
		case directoryActionTypes.FETCH_DIRECTORY_SECTIONS_SUCCESS:
			return {
				...state,
				sections: action.payload,
			}
		case directoryActionTypes.FETCH_DIRECTORY_SECTIONS_FAILURE:
			return {
				...state,
				errorMessage: action.payload,
			}
		default:
			return state
	}
}

export default directoryReducer
