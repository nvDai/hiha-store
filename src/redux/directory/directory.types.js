const directoryActionTypes = {
	FETCH_DIRECTORY_SECTIONS_START: 'FETCH_DIRECTORY_SECTIONS_START',
	FETCH_DIRECTORY_SECTIONS_SUCCESS: 'FETCH_DIRECTORY_SECTIONS_SUCCESS',
	FETCH_DIRECTORY_SECTIONS_FAILURE: 'FETCH_DIRECTORY_SECTIONS_FAILURE',
}
export default directoryActionTypes
