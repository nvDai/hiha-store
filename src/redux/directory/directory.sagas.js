import { call, all, takeLatest, put } from 'redux-saga/effects'
import directoryActionTypes from './directory.types'
import {
	fetchDirectorySectionSuccess,
	fetchDirectorySectionFailure,
} from './directory.actions'
import { firestore } from '../../firebase/firebase.utils'
import { convertSnapshotToSectionsArray } from './directory,utils'

export function* fetchDirectorySections() {
	try {
		const directoryRef = firestore.collection('directory')
		const snapshot = yield directoryRef.get()
		const sections = yield call(convertSnapshotToSectionsArray, snapshot)
		yield put(fetchDirectorySectionSuccess(sections))
	} catch (error) {
		yield put(fetchDirectorySectionFailure(error.message))
	}
}

export function* onFetchDirectorySectionsStart() {
	yield takeLatest(
		directoryActionTypes.FETCH_DIRECTORY_SECTIONS_START,
		fetchDirectorySections
	)
}

export function* directorySaga() {
	yield all([call(onFetchDirectorySectionsStart)])
}
