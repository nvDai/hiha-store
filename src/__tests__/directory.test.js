import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Directory } from '../components/directory/directory.component'

describe('Directory', () => {
	it('should render without crashing', () => {
		const props = {
			sections: [
				{
					id: 1,
					imageUrl: 'https://ik.imagekit.io/fireman35/images/cvpntL1/hats.png',
					linkUrl: 'shop/hats',
					title: 'hats',
				},
				{
					id: 2,
					imageUrl:
						'https://ik.imagekit.io/fireman35/images/px2tCc3/jackets.png',
					linkUrl: 'shop/jackets',
					title: 'jackets',
				},
				{
					id: 3,
					imageUrl:
						'https://ik.imagekit.io/fireman35/images/0jqHpnp/sneakers.png',
					linkUrl: 'shop/sneakers',
					title: 'sneakers',
				},
				{
					id: 4,
					imageUrl:
						'https://ik.imagekit.io/fireman35/images/GCCdy8t/womens.png',
					linkUrl: 'shop/womens',
					size: 'large',
					title: 'womens',
				},
				{
					id: 5,
					imageUrl: 'https://ik.imagekit.io/fireman35/images/R70vBrQ/men.png',
					linkUrl: 'shop/mens',
					size: 'large',
					title: 'mens',
				},
			],
		}
		const wrapper = shallow(<Directory {...props} />)

		expect(toJson(wrapper)).toMatchSnapshot()
	})
})
