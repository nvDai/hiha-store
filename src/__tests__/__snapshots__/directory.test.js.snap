// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory should render without crashing 1`] = `
<div
  className="directory-menu"
>
  <withRouter(MenuItem)
    imageUrl="https://ik.imagekit.io/fireman35/images/cvpntL1/hats.png"
    key="1"
    linkUrl="shop/hats"
    title="hats"
  />
  <withRouter(MenuItem)
    imageUrl="https://ik.imagekit.io/fireman35/images/px2tCc3/jackets.png"
    key="2"
    linkUrl="shop/jackets"
    title="jackets"
  />
  <withRouter(MenuItem)
    imageUrl="https://ik.imagekit.io/fireman35/images/0jqHpnp/sneakers.png"
    key="3"
    linkUrl="shop/sneakers"
    title="sneakers"
  />
  <withRouter(MenuItem)
    imageUrl="https://ik.imagekit.io/fireman35/images/GCCdy8t/womens.png"
    key="4"
    linkUrl="shop/womens"
    size="large"
    title="womens"
  />
  <withRouter(MenuItem)
    imageUrl="https://ik.imagekit.io/fireman35/images/R70vBrQ/men.png"
    key="5"
    linkUrl="shop/mens"
    size="large"
    title="mens"
  />
</div>
`;
